sdk/identity/azure-identity/inc/azure/identity/azure_pipelines_credential.hpp

```diff
@@ -90,6 +90,20 @@ namespace Azure { namespace Identity {
         std::string systemAccessToken,
         AzurePipelinesCredentialOptions const& options = {});
 
+    /**
+     * @brief Constructs an Azure Pipelines Credential using the following environment variables.
+     * AZURESUBSCRIPTION_TENANT_ID, AZURESUBSCRIPTION_CLIENT_ID,
+     * AZURESUBSCRIPTION_SERVICE_CONNECTION_ID, SYSTEM_ACCESSTOKEN
+     *
+     * @param options Options for token retrieval.
+     */
+    explicit AzurePipelinesCredential(AzurePipelinesCredentialOptions const& options = {})
+        : AzurePipelinesCredential(
```

Comment:
We cannot add such an API with these default env vars.

Sharing the gist some the technical details/design context for such a decision: None of the languages do this today, and the env vars may not be stable. We're waiting until a cleaner solution is available as stable by the service, before exposing such conveniences.

Reply:
I believe this is a mistake by the Identity working group. I spoke with Christopher in length about this. The belief that a user passes them verse after reading from the Env rather than the credential reading from the Env is really passing the buck and provides no security benefit.